components/layout/layout_impl.rs

```diff
@@ -253,16 +254,26 @@ impl Layout for LayoutThread {
             .remove_all_web_fonts_from_stylesheet(&stylesheet);
     }
 
+    /// Return a node's bounding box in physical coordinates corresponding to it's Document
+    /// viewport coordinate space. Variation of [`LayoutThread::query_content_boxes`]
+    /// that returns union of it.
+    ///
+    /// Part of <https://drafts.csswg.org/cssom-view-1/#element-get-the-bounding-box>
+    /// TODO(stevennovaryo): Rename and parameterize the function, allowing padding area
+    ///                      query and possibly, query without consideration of transform.
     #[servo_tracing::instrument(skip_all)]
     fn query_content_box(&self, node: TrustedNodeAddress) -> Option<UntypedRect<Au>> {
         let node = unsafe { ServoLayoutNode::new(&node) };
-        process_content_box_request(node)
+        self.process_post_composite_query(node, process_content_box_request)
     }
 
+    /// boxes are in in physical coordinates and correspond to it's Document
+    ///
+    /// Part of <https://drafts.csswg.org/cssom-view-1/#element-get-the-bounding-box>
     #[servo_tracing::instrument(skip_all)]
     fn query_content_boxes(&self, node: TrustedNodeAddress) -> Vec<UntypedRect<Au>> {
         let node = unsafe { ServoLayoutNode::new(&node) };
-        process_content_boxes_request(node)
+        self.process_post_composite_query(node, process_content_boxes_request)
```

Comment:
Instead of having the details of the rectangle calculation here, please pass in the StackingContextTree here as an argument to process_content_boxes_request.

Reply:
Got it, thanks!